sfcasts/crud-controller.md

```diff
@@ -1,171 +1,90 @@
+
+By the time this method is called, it will pass us. The actions of that was already set up by our dashboard, so it will already have the `DETAIL` action enabled for the index page. But *now*, we can control that by saying:
+
+```terminal
+->disable(Action::DETAIL)
+```
+
+We're gonna talk more about the actions configuration, but these are the main things that you can do inside of them: You can add a new action to a page, *or* you can just completely disable an action globally for all of your pages. As soon as we refresh here, our `DETAIL` action is gone, but if we go to our other sections, it's *still* there.
+
+The big takeaway here is that everything is going through our `DashboardController`, which means that we can configure things on a dashboard level, which will apply to all of our CRUD, *or* we can configure things for one specific CRUD. The fact that all of the CRUD controllers go through this `/avenue`, where all has one other effect related to security, means that all of our controllers are already secure. That's thanks to our `access_control`.
+
+Remember back in `config/packages/security.yaml`, we had a little access control here that said if the URL starts with `/admin`, require `roll_admin`. So all of these automatically, without doing anything else, require `role_admin`. We'll talk more later about how to secure different admin controllers with different roles, but at the very least, you have to have `role_admin` to get anywhere, which is awesome.
+
+But one important point: Adding this `access_control` was necessary. Why? Even though the URL is `/admin`, when you go to a `crudController` like this, it does not execute our `DashboardController` index action. This can be a little confusing.
+
+This action here is what has our route. When we go to a `crudController` like this, it does match this route, but EasyAdmin does something crazy. Instead of allowing Symfony to call this controller, it sees this `crudController` query parameter here and magically switches the controller to be the real controller - in this case, to be `QuestionCrudController` index. You can actually see this down here on the web debug toolbar. If you hover over "@admin", this tells you that the route name is "admin". So the route name is *actually* this route right here, but the controller is `QuestionCrudController::index`. There's some route magic happening, but ultimately the methods in your `CrudController` are the ones that are called. It's the index method in this abstract `CrudController` down here, which is the actual roller for the page. This is important for security, because if we had only put the route and the route attribute above index and not added the access control, that wouldn't have been enough. This `isGranted` attribute is only enforced when you actually go to your dashboard page.
```

Comment:
i just realized that there's a duplicate instance of "go to a `crudController" like this..." (one in the previous paragraph too).

Also, the ticks for `crudController` didn't really do anything format-wise here. Does it need them?

And also also, I realized I failed to be consistent with capitalizing "crud" in "crudController". I'm a mess...

Reply:
I see the fancy ticks aren't highlighting any other terms in this paragraph either. I'm sure I messed something up, but I'm not sure where...

Reply:
> i just realized that there's a duplicate instance of "go to a `crudController" like this..." (one in the previous paragraph too).

Yup, no worries. I'm a little confused here - I was having trouble explaining something. So, do your best, but if you don't really get what I'm trying to say, doing less formatting (and leaving more of the original text) is better.

> Also, the ticks for crudController didn't really do anything format-wise here. Does it need them?

Nope! "CRUD controller" is what to use

> I see the fancy ticks aren't highlighting any other terms in this paragraph either. I'm sure I messed something up, but I'm not sure where...

Yea, that's weird! But I don't see the issue - if there is a problem, my editor will show me later.

Reply:
I'm going to be honest. I have no idea how to re-write this but I'll give it a shot. I may try to back up the similar comments as a leaping off point. That means cutting out a sentence or two. I'm not really sure what info is super necessary, and what can be cut/re-worded, so I'll try to mark this paragraph or two for a second review.